Mathlib/Data/Finsupp/Antidiagonal.lean

```diff
@@ -50,6 +50,16 @@ theorem mem_antidiagonal {f : α →₀ ℕ} {p : (α →₀ ℕ) × (α →₀
     ← Multiset.toFinsupp_eq_iff (f := f)]
 #align finsupp.mem_antidiagonal Finsupp.mem_antidiagonal
 
+/-- Finsupp.antidiagonal coincides with Finset.antidiagonalOfLocallyFinite.antidiagonal -/
+lemma antidiagonal_eq_antidiagonal (f : α →₀ ℕ) :
+    antidiagonal f = Finset.antidiagonalOfLocallyFinite.antidiagonal f := by
+  ext p
+  simp only [mem_antidiagonal, Finset.antidiagonalOfLocallyFinite.mem_antidiagonal]
+
+/-- Antidiagonal for Nat valued functions with finite support -/
+abbrev HasAntidiagonal : Finset.HasAntidiagonal (α →₀ ℕ) :=
+  ⟨antidiagonal, @mem_antidiagonal α _⟩
```

Comment:
This one should be an `instance`, and it should be *instead* of the `antidiagonal` definition above. The idea here is to unify all the antidiagonals under a single function name (the one from the typeclass).

Reply:
A question : `mem_antidiagonal` was tagged with `@[simp]`. How do restore this behaviour?

Reply:
`attribute [simp] mem_antidiagonal` immediately after the `export` line